include/seastar/util/modules.hh

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2023 ScyllaDB
+ */
+
+#pragma once
+
+#ifndef SEASTAR_MODULE_EXPORT
+#  define SEASTAR_MODULE_EXPORT
+#  define SEASTAR_MODULE_EXPORT_BEGIN
+#  define SEASTAR_MODULE_EXPORT_END
+#  define SEASTAR_BEGIN_INTERNAL_NAMESPACE namespace internal {
+#  define SEASTAR_END_INTERNAL_NAMESPACE }
+#endif
```

Comment:
Where is the #else?

Reply:
the `#else` branch is in `seastar.cc`. i don't want to pretend that these macros are defined multi times in different translation units. i want to make it very clear that: otherwise, it is define in a single place for only once.

Reply:
if you think a symmetric definition is better, we can have something like
```c++
#ifdef SEASTAR_MODULE
#  define SEASTAR_MODULE_EXPORT export
#  define SEASTAR_MODULE_EXPORT_BEGIN export {
#  define SEASTAR_MODULE_EXPORT_END }
#  define SEASTAR_BEGIN_INTERNAL_NAMESPACE \
    }                                    \
    namespace internal {
#  define SEASTAR_END_INTERNAL_NAMESPACE   \
    }                                    \
    export {
#else 
#  define SEASTAR_MODULE_EXPORT
#  define SEASTAR_MODULE_EXPORT_BEGIN
#  define SEASTAR_MODULE_EXPORT_END
#  define SEASTAR_BEGIN_INTERNAL_NAMESPACE namespace internal {
#  define SEASTAR_END_INTERNAL_NAMESPACE }
#endif
```

but this makes the preprocessed header file a little bit bigger.

Reply:
applied the change in the latest revision.